eth/logmonitor.go

```diff
@@ -43,8 +43,10 @@ func NewLogMonitor(eth LivepeerEthClient, broadcasterAddr, transcoderAddr common
 				job.StreamId = strmID
 				job.TranscodingOptions = transOptions
 
-				for _, cb := range m.callbacks {
-					cb(job)
+				if eth.IsAssignedTranscoder(job.MaxPricePerSegment) {
```

Comment:
Noting here that if the transcoder is assigned to a job, it will need to make the first call to `claimWork` within 256 blocks (due to the use of `block.blockhash` in the contract) of the claim block to explicitly set itself as the assigned transcoder for a job. After the first call, all subsequent calls to `claimWork` can occur at any time. We'll need to change the logic for how a call to `claimWork` is made - I believe right now it happens when a stream is finished, so we would need to change it such that the first call happens within 256 blocks of the claim block and then perhaps a second call when the stream is finished

Reply:
Ahh right.  I'll make that change.